src/ethereum_test_fixtures/blockchain.py

```diff
@@ -455,6 +478,33 @@ class BlockchainFixture(BlockchainFixtureCommon):
     blocks: List[FixtureBlock | InvalidFixtureBlock]
     seal_engine: Literal["NoProof"] = Field("NoProof")
 
+    @field_validator("blocks", mode="before")
+    @classmethod
+    def assign_fork(
+        cls, blocks: List[FixtureBlock | InvalidFixtureBlock], info: ValidationInfo
+    ) -> List[FixtureBlock | InvalidFixtureBlock]:
+        """Assign fork to blocks for model validation."""
+        for block in blocks:
+            if isinstance(block, dict) and "blockHeader" in block:
+                header = block.get("blockHeader")
+                if isinstance(header, dict):
+                    network = info.data.get("fork")
+                    if network == "Frontier":
+                        header["fork"] = Frontier
+                    elif network == "Homestead":
+                        header["fork"] = Homestead
+                    elif network == "Constantinople":
+                        header["fork"] = Constantinople
+                    elif network == "Istanbul":
+                        header["fork"] = Istanbul
+                    elif network == "Paris":
+                        header["fork"] = Paris
+                    elif network == "Cancun":
+                        header["fork"] = Cancun
+                    elif network == "Prague":
+                        header["fork"] = Prague
```

Comment:
```suggestion
                    if network in all_forks_by_name:
                        header["fork"] = all_forks_by_name[network]
```
Follow up to the previous comment.